### ISSUE TITLE
WebSocket client

### ISSUE THREAD

**Issue by trustin** [line/armeria#4234]
@minwoox's PR #3904 adds the server-side WebSocket support. It'd be awesome if we can also add the client-side WebSocket support, which could be useful for those who implements a WebSocket gateway.

Hypothetical API:

```java
WebSocketClient client = WebSocketClient.of(); // or WebSocketClient.of("https://example.com/ws")

// Perhaps we could introduce an event driven API instead of this?
StreamMessageWriter<WebSocketFrame> out = StreamMessage.streaming();
StreamMessage<WebSocketFrame> in = client.connect(out);
```

**Comment by i1619khz** [comment]
let me try

**Comment by minwoox** [comment]
Thanks for volunteering for this issue. 👍 
I'm working on implementing [WebSocket service](https://github.com/line/armeria/pull/3904) and there are common classes 
used both for the WebSocket client and service. Let me finish the service first and ping you so that you can work with those classes. 😄

**Comment by i1619khz** [comment]
ok

**Comment by minwoox** [comment]
Hi, @i1619khz! I'm so sorry for taking away your joy but I had to implement this by myself.
https://github.com/line/armeria/pull/4972
After implementing `WebSocketService`, I need to also implement the client in order to check if they work correctly and modify both sides of APIs if needed. 🙏 
Please, let me know if you are interested in implementing another feature. I might give you some help to find it.

**Comment by ikhoon** [comment]
Closed by #4972
